### ISSUE TITLE
Provide better error analysis when Service Bus messages cannot be deserialized

### ISSUE THREAD

**Issue by brettsam** [Azure/azure-webjobs-sdk#897]
We've have a decent number of people run into vague SB deserialization errors that are tough to understand. We should be able to write out some better error messages when we encounter something like this -- maybe even printing out the ContentType, etc. Anything that points them in the right direction for getting the messages formatted correctly. Right now that tends to mean hunting through github issues or stackoverflow.

I'll use this issue to collect the scenarios we've seen so we can improve this.

**Comment by seantaylorwolterskluwer** [comment]
I agree with Brett. I am having a terrible figuring why I am getting the following error. I dont see any logs anywhere that would help me debug this.

`Exception binding parameter 'myQueueItem'. System.Runtime.Serialization: There was an error deserializing the object of type System.String. The input source is not correctly formatted.`
